counsel.el

```diff
@@ -2063,6 +2063,59 @@ By default `counsel-bookmark' opens a dired buffer for directories."
    ("r" ,(counsel--apply-bookmark-fn #'counsel-find-file-as-root)
         "open as root")))
 
+;;** `counsel-bookmarked-directory'
+(defun counsel-bookmarked-directory--candidates ()
+  "Get a list of bookmarked directories sorted by file path."
+  (bookmark-maybe-load-default-file)
+  (sort (cl-remove-if-not
+         #'ivy--dirname-p
+         (mapcar #'bookmark-get-filename bookmark-alist))
+        #'string<))
+
+;;;###autoload
+(defun counsel-bookmarked-directory-add (dir)
+  "Create a directory bookmark to DIR interactively.
+
+This function prompts for the bookmark name, which defaults to the
+abbreviate path of the directory.
+
+When called-interactively, DIR defaults to he current value of
+`default-directory'."
+  (interactive (list default-directory))
+  (let* ((abbr-path (abbreviate-file-name (expand-file-name dir)))
+         (default-name abbr-path)
+         (name (read-string (format "Name of the new directory bookmark [%s]: "
+                                    default-name)
+                            nil nil default-name))
```

Comment:
OK, I understand now that this is just meant to be a bookmark name, rather than a directory, which is why `read-directory-name` isn't used. Either way, no need for both `abbr-path` and `default-name`; pick a single descriptive variable name, e.g. `abbr-dir`, and use that consistently. In fact, I would reuse the argument `dir`:

```el
(setq dir (abbreviate-file-name (expand-file-name dir)))
(let ((name (read-string (format "Name of new directory bookmark [%s]: " dir)
                         nil nil dir))
      (record `((filename . ,(file-name-as-directory dir))
                (front-context-string)
                (rear-context-string))))
  (bookmark-maybe-load-default-file)
  (bookmark-store name record nil))
```